src/Food101.jl

```diff
@@ -47,27 +47,48 @@ Base.show(io::IO, X::Food101Targets) = print(io, "Food101Targets()")
 
 ## data
 
-function getobs(data::Food101Data, idx::Integer)
+function getobs!(buf::Array{UInt8,3}, data::Food101Data, idx::Integer)
     ds = data.dataset
     path = joinpath(ds.dir, "images", ds.files[idx] * ".jpg")
-    return permutedims(rawview(channelview(im)), (2,3,1))    
+    im_ = nothing
+    try
+        # println(idx)
+        im_ = load(path)
+        im = imresize(im_, (512, 512))
+        buf .= permutedims(rawview(channelview(im)), (2,3,1))
+    catch e
+        warn("Can't load $path")
+        buf .= zeros(UInt8, 512, 512, 3)
+    end    
+    return buf
 end
 
 
-function getobs(data::Food101Data, idxs::Union{Vector{<:Integer}, UnitRange{<:Integer}})
-    ret = Array{UInt8,4}(512, 512, 3, length(idxs))
-    for idx in idxs
-        ret[:,:,:,idx] = getobs(data, idx)
+function getobs!(buf::Array{UInt8,4}, data::Food101Data,
+                 idxs::Union{Vector{<:Integer}, UnitRange{<:Integer}})
+    for (i, idx) in enumerate(idxs)
+        
+        getobs!(buf[:,:,:,i], data, idx)
     end
-    return ret
+    return buf
 end
 
 
+getobs(data::Food101Data, idx::Integer) = getobs!(Array{UInt8}(512,512,3), data, idx)
+
+getobs(data::Food101Data, idxs::Union{Vector{<:Integer}, UnitRange{<:Integer}}) =
```

Comment:
`::Union{Vector{<:Integer}, UnitRange{<:Integer}}`

is there a reason to be so specific instead of using `::AbstractVector{<:Integer}`

Reply:
Haven't thought about it. Thanks!